src/overlays/actors/ovl_En_Time_Tag/z_en_time_tag.h

```diff
@@ -7,10 +7,57 @@ struct EnTimeTag;
 
 typedef void (*EnTimeTagActionFunc)(struct EnTimeTag*, PlayState*);
 
-#define ENTIMETAG_GET_1F(thisx) ((thisx)->params & 0x1F)
-#define ENTIMETAG_GET_SWITCHFLAG(thisx) ((thisx)->params & 0x7F)
-#define ENTIMETAG_GET_1FE0(thisx) (((thisx)->params & 0x1FE0) >> 0x5)
-#define ENTIMETAG_GET_E000(thisx) (((thisx)->params & 0xE000) >> 0xD)
+#define TIMETAG_GET_TYPE(thisx) (((thisx)->params & 0xE000) >> 0xD)
+
+#define TIMETAG_GET_KICKOUT_TEXT_TYPE(thisx) (((thisx)->params & 0x1FE0) >> 0x5)
+#define TIMETAG_GET_KICKOUT_EXIT_INDEX(thisx) ((thisx)->params & 0x1F)
+#define TIMETAG_KICKOUT_HOUR(thisx) ((thisx)->home.rot.x)
+#define TIMETAG_KICKOUT_MINUTE(thisx) ((thisx)->home.rot.y)
+
+#define TIMETAG_GET_SOARING_SWITCHFLAG(thisx) ((thisx)->params & 0x7F)
+
+#define TIMETAG_GET_DIARY_TYPE(thisx) ((thisx)->params & 0x7F)
+#define TIMETAG_GET_DIARY_TIMER(thisx) ((thisx)->home.rot.x)
+#define TIMETAG_GET_DIARY_SONG(thisx) ((thisx)->home.rot.z)
+
+typedef enum {
+    /* 0 */ TIMETAG_KICKOUT_DOOR,
+    /* 1 */ TIMETAG_SOARING_ENGRAVING,
+    /* 2 */ TIMETAG_ROOFTOP_OATH,
+    /* 3 */ TIMETAG_DIARY,
+    /* 4 */ TIMETAG_KICKOUT_FINAL_HOURS,
+    /* 5 */ TIMETAG_KICKOUT_NO_DOOR
+} TimeTagType;
+
+typedef enum {
+    /* 0 */ DIARY_MIKAU,
+    /* 1 */ DIARY_LULU
+} DiaryType;
+
+typedef enum {
+    /* 0 */ DIARY_SONG_EVAN_PART1,
+    /* 1 */ DIARY_SONG_EVAN_PART2
+} DiarySong;
+
+typedef enum {
+    /*  0 */ KICKOUT_TEXT_TYPE_0, // I am closing the shop
+    /*  1 */ KICKOUT_TEXT_TYPE_SWAMP_SHOOTING_GALLERY, // time to close shop
+    /*  2 */ KICKOUT_TEXT_TYPE_STOCK_POT_INN, // stock pot in locking the door
+    /*  3 */ KICKOUT_TEXT_TYPE_HONEY_AND_DARLING, // need some alone time
+    /*  4 */ KICKOUT_TEXT_TYPE_LOTTERY, // business is finished
+    /*  5 */ KICKOUT_TEXT_TYPE_BARN_DAY, // the lady needs a break
+    /*  6 */ KICKOUT_TEXT_TYPE_DEFAULT, // (Many scenes) place is about to close
+    /*  7 */ KICKOUT_TEXT_TYPE_BARN_NIGHT, // they came
+    /*  8 */ KICKOUT_TEXT_TYPE_TRADING_POST, // we have to close shop
+    /*  9 */ KICKOUT_TEXT_TYPE_MILK_BAR, // I have to get ready for business
+    /* 10 */ KICKOUT_TEXT_TYPE_CURIOSITY_SHOP_BACK_1, // shop is opening, you need to go
+    /* 11 */ KICKOUT_TEXT_TYPE_CURIOSITY_SHOP_BACK_2, // shop is opening, we're in the way
+    /* 12 */ KICKOUT_TEXT_TYPE_TOWN_SHOOTING_GALLERY_1, // it's time to close
+    /* 13 */ KICKOUT_TEXT_TYPE_TOWN_SHOOTING_GALLERY_2, // quick, it's closing time
+    /* 14 */ KICKOUT_TEXT_TYPE_SWORDSMANS_SCHOOL, // need to cut the moon, must leave
+    /* 15 */ KICKOUT_TEXT_TYPE_TREASURE_CHEST_SHOP, // already time to close shop
+    /* 16 */ KICKOUT_TEXT_TYPE_DOGGY_RACETRACK, // no more for today
+} KickOutTextType;
```

Comment:
Probably still a good idea to prefix with `TimeTag` here to help prevent possible naming collisions.